app.py

```diff
@@ -939,38 +982,81 @@ def snap_release(snap_name):
 @app.route('/account/snaps/<snap_name>/market/', methods=['POST'])
 @login_required
 def post_market_snap(snap_name):
-    whitelist = [
-        'title',
-        'summary',
-        'description',
-        'contact_url',
-        'keywords',
-        'license',
-        'price',
-        'blacklist_countries',
-        'whitelist_countries'
-    ]
-
     if 'submit_revert' in flask.request.form:
         flask.flash("All changes reverted.", 'information')
     else:
+        error_list = []
+        info = []
+        files = []
+        images_json = None
+
+        icon = flask.request.files.get('icon')
+        if icon is not None:
+            info.append(build_image_info(icon, 'icon'))
+            files.append(icon)
+
+        screenshots = flask.request.files.getlist('screenshots')
+        for screenshot in screenshots:
+            info.append(build_image_info(screenshot, 'screenshot'))
+            files.append(screenshot)
+
+        if not files:
+            # API requires a multipart request, but we have no files to push
+            # https://github.com/requests/requests/issues/1081
+            files = {'info': ('', dumps(info))}
+        else:
+            images_json = {'info': dumps(info)}
+
+        screenshots_list = snap_screenshots(
+            flask.request.form['snap_id'],
+            images_json,
+            files
+        )
+
+        if 'error_list' in screenshots_list:
+            error_list.append(
+                screenshots_list['error_list']
+            )
+
+        whitelist = [
+            'title',
+            'summary',
+            'description',
+            'contact_url',
+            'keywords',
+            'license',
+            'price',
+            'blacklist_countries',
+            'whitelist_countries'
+        ]
+
         body_json = {
             key: flask.request.form[key]
             for key in whitelist if key in flask.request.form
         }
 
         metadata = snap_metadata(flask.request.form['snap_id'], body_json)
-
         if 'error_list' in metadata:
+            error_list.append(
```

Comment:
I'm not sure if that's correct.
It seem to append an array of errors INTO an array, making error_list contain another array instead of list of errors, like:

`[[{'message': 'Image must be exactly 256x256 pixels in size.', 'extra': {'name': 'icon'}, 'code': 'invalid-field'}]]`